### PR TITLE
modules: tt.yaml modules/directory support list

### LINKED ISSUE #1012: `modules:` extend `tt.yaml` to support a list in `modules/directory`
The logic of working with the configuration file tt.yaml should be updated, allowing to specify modules/directory both as a single line and as a list.

By default, should be a single entry:
```yaml
modules:
  directory: /path/to/ext/tt/modules
```

And it should accept a list of directories:
```yaml
modules:
  directory:
    - modules
    - /path/to/ext/tt/modules
```

If a relative path is specified, the search is performed in the subfolders below, relative to the `tt.yaml` file. If an absolute path is specified, external modules are searched according to the specified path.
When creating a new configuration file, the default configuration should have only the single `modules` folder is defined.

### PR DESCRIPTION
**Pull request by dmyger**
Closes #1012

@TarantoolBot document
Title: Allow multiple directories with modules

The logic of working with configuration file tt.yaml has been updated,
now  modules/directory can be specified both as a single line and
as a list.

**Usage example**
Old behavior:
```yaml
modules:
  directory: modules
```
Possibility to specify a list of directories:
```yaml
modules:
  directory:
  - modules
  - /ext/path/modules
  - other_modules
```

If a relative path is specified, the search is performed in
the subfolders below, relative to the tt.yaml file. If an absolute
path is specified, external modules are searched according to
the specified path.

### CODE REVIEW

File: cli/config/single_or_array_test.go
Comment (patapenka-alexey) [inline cli/config/single_or_array_test.go:23]: Maybe, you forgot to remove this?

File: cli/config/single_or_array_test.go
Comment (patapenka-alexey) [inline cli/config/single_or_array_test.go:154]: And here.

**Comment by oleg-jukovec** [comment]
Please, rebase on the master branch and change a target.

File: cli/config/single_or_array_test.go
Comment (oleg-jukovec) [inline cli/config/single_or_array_test.go:37]: It will be on error message.
```suggestion
```

File: cli/config/single_or_array_test.go
Comment (oleg-jukovec) [inline cli/config/single_or_array_test.go:169]: The same.
```suggestion
```

**Review by oleg-jukovec — APPROVED** [review]

**Comment by oleg-jukovec** [comment]
But let's wait for a green tests before a merge.

File: cli/config/single_or_array_test.go
Comment (dmyger) [reply cli/config/single_or_array_test.go:23]: Thanks, it was commented out just for debugging.

File: cli/config/single_or_array_test.go
Comment (dmyger) [reply cli/config/single_or_array_test.go:37]: Removed.

File: cli/config/single_or_array_test.go
Comment (dmyger) [reply cli/config/single_or_array_test.go:154]: Done.

File: cli/config/single_or_array_test.go
Comment (dmyger) [reply cli/config/single_or_array_test.go:169]: Removed.

**Review by oleg-jukovec — APPROVED** [review]

**Pull request merged**
